app/src/ui/diff/diff-syntax-mode.ts

```diff
@@ -166,7 +205,19 @@ export class DiffSyntaxMode {
         return null
       }
 
-      let result = getBaseDiffLineStyle(token)
+      // Hack to get the next line. We found that the StringStream has an extra
+      // property of lookAhead at run time that has what we need to access the
+      // next line. This just exposes it so we can use it.
+      const nextLine = (stream as StringStreamWithLookahead).lookAhead(1)
+      const nextLineTokenIndex = nextLine !== undefined ? nextLine[0] : null
```

Comment:
This type is lying a bit, it's defined as `string | null` but it can actually be `string | null | undefined`. I think it would make sense to redefine the type to be `string | undefined` and change this to

```suggestion
      const nextLineTokenIndex =
        typeof nextLine === 'string' ? nextLine[0] : undefined
```